corehq/apps/smsbillables/generator.py

```diff
@@ -150,7 +151,12 @@ def arbitrary_fees_by_all(backend_ids):
 def arbitrary_backend_ids():
     backend_ids = {}
     for backend in get_available_backends().values():
-        backend_ids[backend.get_api_id()] = data_gen.arbitrary_unique_name("back")
+        backend_instance = data_gen.arbitrary_unique_name("back")
+        backend_ids[backend.get_api_id()] = backend_instance
+        sms_backend = SMSBackend()
```

Comment:
Do you ever need the backends to be of their subclass type in the tests? If so, you'll have to instantiate this with backend() rather than SMSBackend(). If not, then it's actually good to leave it like this because calling SMSBackend.send() raises an exception rather than trying to connect with the gateway.